### PR TITLE
NO-ISSUE: add a separate makefile for code generation

### PR DESCRIPTION
**Pull request by osherdp**
Currently we are heavily dependant on the ``hack/generate.sh`` file for all out file generation tasks. It has several drawbacks:

* Available generation options are not known beforehand, meaning one cannot use makefile autocompletion in order to understand what are the available options.
* ``skipper make generate`` has been adapted to print usage message, which is kind of weird for a target with such action name in it.
* Some (usefull) targets like ``lint-swagger`` are not exposed for the user.

This change aims to add a new Makefile relevant to all files in ``hack/`` directory, including it in the main Makefile. All targets
which are less than 3 lines long are written inline, and the other ones will stay in ``hack/generate.sh``.

The ``generate-all`` target will stay for backwards-compatibility, but eventually be replaced by ``generate``.

## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @eliorerz @filanov @celebdor 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: hack/Makefile [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	./hack/generate.sh generate_crds
+
+generate-bundle:
+	./hack/generate.sh generate_bundle
+
+generate-from-swagger: lint-swagger generate-go-client generate-go-server validate-swagger-file remove-dashes-and-dots
+
+generate: generate-from-swagger generate-events generate-mocks generate-configuration
+
+generate-all: generate # backwards compatibility
```
Comment (eliorerz) [inline hack/Makefile:52]: Do we have a task for removing it in the future?

File: hack/Makefile [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+generate-keys: ${BUILD_FOLDER}
+	(cd ./tools && go run auth_keys_generator.go -keys-dir=${BUILD_FOLDER})
+
+generate-events:
+	rm -rf internal/common/events
+	mkdir -p internal/common/events
+	tools/generate_events.py ./docs/events.yaml internal/common/events/events.go
+
+remove-dashes-and-dots:
```
Comment (eliorerz) [inline hack/Makefile:30]: Maybe we can make it private target (`--remove-dashes-and-dots`)?

File: hack/Makefile (hunk 1 above)
Comment (osherdp) [reply hack/Makefile:52]: https://issues.redhat.com/browse/MGMT-10029

File: hack/Makefile (hunk 2 above)
Comment (osherdp) [reply hack/Makefile:30]: sure why not

**Review by eliorerz — APPROVED** [review]
LGTM

**Comment by osherdp** [comment]
/test e2e-metal-assisted-onprem
now presubmit jobs are fixed

**Comment by osherdp** [comment]
All seems to work pretty well
/unhold

**Pull request merged**